synapse/cortex.py

```diff
@@ -1454,16 +1454,17 @@ async def _storLayrFeedDefaults(self):
                 await layrinfo.set('pulls', pulls, nexs=False)
 
     async def initServiceRuntime(self):
```

Comment:
Would it be easier to add a new `runServiceMigrations()` method ( called by the Cell right before `initServiceRuntime` ) to have a dedicated step between nexus init and the rest of service junk coming online?

Reply:
The problem is that we need the service migrations to run before the cortex goes active (via `setCellActive()`) because `setCellActive(True)` is where the view and layer tasks get started (and subsequently stopped when `setCellActive(False)`).

I can't just wedge in `runServiceMigrations()` before `setCellActive()` in `initNexusSubsystem()` either because we need `_initCoreMods()` (and maybe some others that I haven't nailed down yet) to run from `initServiceRuntime()` *before* we run the migrations otherwise the migrations will be incomplete.

I'm basically having to shuffle the entire cortex (and maybe the cell too?) startup sequence to get these migrations in the right place.